### ISSUE TITLE
Flash or YouTube videos fullscreen also maximize browser.

### ISSUE THREAD

**Issue by TheGrandFinale2001** [compiz-reloaded/compiz#115]
I like to keep my windows tiled to either side of my screen.  Whenever I go fullscreen on a flash video or YouTube video, and then exit fullscreen my browser is maximize.  It happens everytime I exit a fullscreen video.

**Comment by ethus3h** [comment]
I suspect this has to do with the windows being maximized horizontally or vertically, but not both, but being restored to both when leaving fullscreen: I can reproduce it when the window is tiled using the Grid plugin in such a way that it spans the whole screen, but not when, for example, it is tiled to a corner.

**Comment by TheGrandFinale2001** [comment]
Yes, you are right.  I just checked and it doesn't do it when windows are tiled to the corners.

**Comment by soreau** [comment]
Here is a [patch](https://gist.github.com/soreau/bb4d9a936af6cf8b9f130cb154027cb1) that fixes the problem by adding a dedicated 'slot' for saving dimensions before going to fullscreen, so you can half-maximize, toggle fullscreen, go back to half-maximize and still snap off to the size before half-maximizing. The patch breaks ABI which means you will need to rebuild all plugins against compiz with this patch applied.

**Comment by TheGrandFinale2001** [comment]
Oh cool, thanks.  I'm on Arch, how would I go about doing this?

**Comment by soreau** [comment]
If you're not comfortable applying a patch, you might be better off waiting until something is pushed to git, so you can just rebuild compiz-reloaded git packages from aur.

**Comment by ethus3h** [comment]
For what it's worth, I just tested this patch and it seems to fix the issue. Thanks @soreau! :tada:

**Comment by soreau** [comment]
Closed with 9752b0cbc
